### PR TITLE
#18837 Catching exception when Es index not exists

### PR DESCRIPTION
**Pull request by freddyucv**
Catching exception when ES index not exists

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 1]
```diff
@@ -1,22 +1,27 @@
 package com.dotcms.util.pagination;
 
+import com.dotcms.content.elasticsearch.business.IndiciesInfo;
 import com.dotcms.contenttype.business.ContentTypeAPI;
 import com.dotcms.contenttype.model.type.BaseContentType;
 import com.dotcms.contenttype.model.type.ContentType;
 import com.dotcms.contenttype.model.type.ContentTypeBuilder;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:8]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.enterprise.rules.RulesAPIImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.rules.model.Condition;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:15]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.Condition'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 3]
```diff
@@ -121,6 +126,38 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+     */
+    @Test
+    public void whenTheIndicesAreDeactivateShouldReturnNAInEntries() throws DotDataException, IOException {
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final String live = indiciesInfo.getLive();
+        final String working = indiciesInfo.getWorking();
+        try {
+            if (live != null) {
+                APILocator.getContentletIndexAPI().deactivateIndex(live.substring(live.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:145]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 4]
```diff
@@ -121,6 +126,38 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+    /**
+     * Method to Test: {@link ContentTypesPaginator#getItems(User, String, int, int, String, OrderDirection, Map)}
+     * When: When the current indices are desactive or deleted
+     * Should: Should return NA in the entries property
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void whenTheIndicesAreDeactivateShouldReturnNAInEntries() throws DotDataException, IOException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:138]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
 import com.dotcms.contenttype.model.type.ContentTypeBuilder;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.rules.model.Condition;
+import com.dotmarketing.portlets.rules.model.ConditionGroup;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:16]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.ConditionGroup'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 6]
```diff
@@ -121,6 +126,38 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final String live = indiciesInfo.getLive();
+        final String working = indiciesInfo.getWorking();
+        try {
+            if (live != null) {
+                APILocator.getContentletIndexAPI().deactivateIndex(live.substring(live.indexOf(".") + 1));
+            }
+
+            APILocator.getContentletIndexAPI().deactivateIndex(working.substring(working.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:148]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java (hunk 3 above)
Comment (jdotcms) [reply dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:145]: agree

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 7]
```diff
@@ -121,6 +126,43 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+
+            final PaginatedArrayList<Map<String, Object>> items = paginator.getItems(user, "", -1, 0);
+
+            for (final Map<String, Object> item : items) {
+                assertEquals("N/A", item.get("nEntries"));
+            }
+
+        }finally {
+            if (live != null) {
+                APILocator.getContentletIndexAPI().activateIndex(live.substring(live.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:160]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:16]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.ConditionGroup'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:15]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.Condition'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 8]
```diff
@@ -121,6 +126,43 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+            }
+
+            APILocator.getContentletIndexAPI().activateIndex(working.substring(working.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:163]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:8]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.enterprise.rules.RulesAPIImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 9]
```diff
@@ -121,6 +126,43 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final String live = indiciesInfo.getLive();
+        final String working = indiciesInfo.getWorking();
+        try {
+            if (live != null) {
+                APILocator.getContentletIndexAPI().deactivateIndex(live.substring(live.indexOf(".") + 1));
+            }
+
+            APILocator.getContentletIndexAPI().deactivateIndex(working.substring(working.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:148]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

File: dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java [hunk 10]
```diff
@@ -121,6 +126,43 @@ public void test_getItems_WhenFilterContainsContentTypeName_ReturnsTheContentTyp
...
+     */
+    @Test
+    public void whenTheIndicesAreDeactivateShouldReturnNAInEntries() throws DotDataException, IOException {
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final String live = indiciesInfo.getLive();
+        final String working = indiciesInfo.getWorking();
+        try {
+            if (live != null) {
+                APILocator.getContentletIndexAPI().deactivateIndex(live.substring(live.indexOf(".") + 1));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:145]: Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5857184)

Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/util/pagination/ContentTypesPaginatorTest.java:145]: For future reference, in the ESIndexAPI there is a method called removeClusterIdFromName that removes the cluster prefix, so you don't have to worry about the substring

**Review by nollymar — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**